From 07a2ec0e86aa016a71d7ca48799f83f5f3663325 Mon Sep 17 00:00:00 2001
From: Eric Naim <dnaim@cachyos.org>
Date: Wed, 30 Apr 2025 02:54:47 +0800
Subject: [PATCH 12/20] Revert "Bluetooth: l2cap: Check encryption key size on
 incoming connection"

This reverts commit d4ca2fd218caafbf50e3343ba1260c6a23b5676a.

This breaks pairing of various bluetooth devices, including DualSense
controllers.

Link: https://bugzilla.kernel.org/show_bug.cgi?id=220061
Signed-off-by: Eric Naim <dnaim@cachyos.org>
---
 net/bluetooth/l2cap_core.c | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/net/bluetooth/l2cap_core.c b/net/bluetooth/l2cap_core.c
index c219a8c59..e9bfaf692 100644
--- a/net/bluetooth/l2cap_core.c
+++ b/net/bluetooth/l2cap_core.c
@@ -3956,8 +3956,7 @@ static void l2cap_connect(struct l2cap_conn *conn, struct l2cap_cmd_hdr *cmd,
 
 	/* Check if the ACL is secure enough (if not SDP) */
 	if (psm != cpu_to_le16(L2CAP_PSM_SDP) &&
-	    (!hci_conn_check_link_mode(conn->hcon) ||
-	    !l2cap_check_enc_key_size(conn->hcon))) {
+	    !hci_conn_check_link_mode(conn->hcon)) {
 		conn->disc_reason = HCI_ERROR_AUTH_FAILURE;
 		result = L2CAP_CR_SEC_BLOCK;
 		goto response;
-- 
2.49.0

